woocommerce/woocommerce
# add support for projects to suppress console output

### Changes proposed in this Pull Request:

Our CI E2E logs are currently running at 2000+ lines due to a large number of deprecated warnings

```
`woocommerce_shared_settings` filter in Blocks is deprecated.
```

Since this wouldn't affect every project using the test runner, it didn't make sense to add filtering this to the package console filtering. 

This PR refactors the console filtering code into 3 exported functions: `addConsoleSuppression`,`removeConsoleSuppression`, and `consoleShouldSuppress`. It also adds the `woocommerce_shared_settings` to the filter list in our local jest setup.

### How to test the changes in this Pull Request:

1. Verify CI Run
2. Run `npm run build:packages`
3. Run E2E tests locally
4. Verify the `woocommerce_shared_settings` is only output once per test file.

### Changelog entry

> N/A
